jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -37,20 +31,46 @@ class FormMedia extends React.Component {
    */
 
   componentDidMount() {
-    this.refreshFormMedia();
+    const callbacks = {
+      onComplete: this.onGetMediaCompleted.bind(this),
+      onFail: this.onGetMediaFailed.bind(this)
+    };
+    actions.media.loadMedia(this.props.asset.uid, callbacks);
     this.setState({isVirgin: false});
   }
 
-  refreshFormMedia() {
-    dataInterface.getFormMedia(this.props.asset.uid).done((uploadedAssets) => {
-      this.setState({
-        uploadedAssets: uploadedAssets.results,
-        isUploadFilePending: false,
-        isUploadURLPending: false
-      });
+  /*
+   * action listeners
+   */
+
+  onGetMediaCompleted(uploadedAssets) {
+    this.setState({
+      uploadedAssets: uploadedAssets.results,
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    });
+  }
+
+  onGetMediaFailed(response) {
+    // Do we need to do more than say 'something went wrong'?
+  }
+
+  onDeleteMediaFailed(response) {
+    // Do we need to do more than say 'something went wrong'?
+  }
```

Comment:
If you don't need to reset anything when it fails, I think it is good enough to have a general error message in `actions.es6`. You can delete these two functions.

Oh and when you delete these, make sure there are no leftover listeners hooked up in `componentDidMount` (e.g. `actions.media.deleteMedia.failed.listen(this.onDeleteMediaFailed);`)